server/genquery2/src/genquery2_sql.cpp

```diff
@@ -674,21 +674,21 @@ namespace
 
         const auto& expressions = _group_by.expressions;
 
-        std::vector<std::string> sort_expr;
-        sort_expr.reserve(expressions.size());
+        std::vector<std::string> group_expr;
+        group_expr.reserve(expressions.size());
 
         for (const auto& expr : expressions) {
             if (const auto* p = std::get_if<irods::experimental::genquery2::column>(&expr); p) {
-                sort_expr.emplace_back(to_sql_for_order_or_group_by_clause(_state, *p));
+                group_expr.emplace_back(to_sql_for_order_or_group_by_clause(_state, *p));
                 continue;
             }
 
             const auto& fn = std::get<irods::experimental::genquery2::function>(expr);
-            sort_expr.emplace_back(to_sql_for_order_or_group_by_clause(_state, fn));
+            group_expr.emplace_back(to_sql_for_order_or_group_by_clause(_state, fn));
         }
 
         // All columns/expressions in the group by clause must exist in the list of columns to project.
-        return fmt::format(" group by {}", fmt::join(sort_expr, ", "));
+        return fmt::format(" group by {}", fmt::join(group_expr, ", "));
```

Comment:
The comment above this line isn't true, as demonstrated in your tests. Please remove it.